server/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -640,14 +634,25 @@ public void close() throws IOException {
     }
 
     /**
-     * @return Whether a field is a metadata field.
+     * @return Whether a field is a metadata field for older indexes
+     * TODO: remove in v 9.0
+     * @deprecated  Use an instance method isMetadataField instead
      */
-    public static boolean isMetadataField(String fieldName) {
-        return META_FIELDS.contains(fieldName);
+    @Deprecated
+    public static boolean isMetadataFieldStatic(String fieldName) {
+        if (IndicesModule.getBuiltInMetadataFields().contains(fieldName)) {
+            return true;
+        }
+        // adding _size field as meta-field for bwc
```

Comment:
Since this sticks out as odd, maybe extend the comment by briefly explaining the cases where we need this. As far as I understand this should only be the case when a pre-7.3 node uses the size mapper plugin and sends these fields jumbled with the other document fields, right?